frontend/mulighetsrommet-veileder-flate/src/hooks/usePrepopulerFilter.ts

```diff
@@ -0,0 +1,39 @@
+import { useAtom } from 'jotai';
+import { useEffect } from 'react';
+import { useHentBrukerdata } from '../api/queries/useHentBrukerdata';
+import { useInnsatsgrupper } from '../api/queries/useInnsatsgrupper';
+import { tiltaksgjennomforingsfilter } from '../core/atoms/atoms';
+
+export function usePrepopulerFilter() {
+  const [filter, setFilter] = useAtom(tiltaksgjennomforingsfilter);
+  const brukerdata = useHentBrukerdata();
+  const { data: innsatsgrupper } = useInnsatsgrupper();
+  const data = brukerdata?.data;
+
+  useEffect(() => {
+    if (data && innsatsgrupper) {
+      forcePrepopulerFilter(false);
+    }
+  }, [data, innsatsgrupper]);
+
+  function forcePrepopulerFilter(resetFilterTilUtgangspunkt: boolean) {
+    const matchedInnsatsgruppe = innsatsgrupper?.find(gruppe => gruppe.tittel === brukerdata?.data?.innsatsgruppe);
+    if (matchedInnsatsgruppe) {
+      const tiltakstyper = resetFilterTilUtgangspunkt ? [] : filter.tiltakstyper;
+      const search = resetFilterTilUtgangspunkt ? '' : filter.search;
+      const innsatsgrupper = resetFilterTilUtgangspunkt
+        ? [{ id: matchedInnsatsgruppe._id, ...matchedInnsatsgruppe }]
+        : [
+            ...filter.innsatsgrupper.filter(gruppe => gruppe.id !== matchedInnsatsgruppe._id),
+            { id: matchedInnsatsgruppe._id, ...matchedInnsatsgruppe },
+          ];
+      setFilter({
+        search,
+        tiltakstyper,
+        innsatsgrupper,
+      });
+    }
+  }
```

Comment:
Dette føles litt rotete så skal prøve å forklare hva som skjer

1. Når vi har hentet innsatsgrupper og brukerdata så prøver vi å finne brukers innsatsgruppe. Vi må finne den fra listen over tiltakstyper slik at vi får korrekt id for filteret
2. Dersom veileder tilbakestiller filteret så blir tiltakstyper til tomt array og search til tom streng. For innsatsgrupper så tar vi enten bare brukers innsatsgruppe, eller så tar vi brukers innsatsgruppe og filter fra url.